- Initial aggregation and preprocess module:

nitieaj opened the pull request:
This module takes the column and class label, groups the column by
  descriptive statistics  and plots associated graphs.

nitieaj committed:
- Initial aggregation and preprocess module:

  This module takes the column and class label, groups the column by
  descriptive statistics  and plots associated graphs.

haleyhowe commented on hs628_tools/Preprocess_module.py:
```diff
@@ -0,0 +1,144 @@
+
+
+
+def coltuple(df,col,typpe):
```
What is the parameter "typpe" referring to ?

nitieaj replied:
The third parameter "typpe" is the column name of the categorical variable e.g class label (mortality).

NikitaThomas commented on hs628_tools/Preprocess_module.py:
```diff
@@ -0,0 +1,144 @@
+# In[245]:
+
+
+def summarystat(listd):
```
I think it would also be helpful to have the min/max values of each column so the user knows the range of each of their variables.

NikitaThomas commented on hs628_tools/Preprocess_module.py:
```diff
@@ -0,0 +1,144 @@
+        plt.rc('grid', linestyle="dashed", color='grey')
+        plt.show()
+        print(i,"Mean = {0}".format(avg),"Median = {0}".format(median),"stdev = {0}".format(stdev))
+    return avg,median,stdev
```
I think it would also be helpful to have the min/max values of each column so the user knows the range of each of their variables.

nitieaj replied:
yeah , that would be a great addition .I am looking to add 5 summary (minimum and maximum number, 1st quartile, median and 3rd quartile).

nitieaj commented:
Yeah, you`re right. I`ll look into adding that component .

On Thu, Jul 26, 2018 at 12:04 PM, Nikita Thomas <notifications@github.com>
wrote:

> *@NikitaThomas* commented on this pull request.
> ------------------------------
>
> In hs628_tools/Preprocess_module.py
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_EHWUSF_HS68-5F2018-5FProject-5F1_pull_32-23discussion-5Fr205569506&d=DwMCaQ&c=qgVugHHq3rzouXkEXdxBNQ&r=0cV_TVOWlRVep9Kq29fJkEZ32-TILX0rx9zYhsQYYog&m=3fZ8LvrpgOvOpQDSEildYnhz_VXywjVQjrXy_4fi618&s=TWuGW3eXPEZYREz8tVeK0WSLppC6GAtF1NwlWLoq8Eg&e=>
> :
>
> > +# In[245]:
> +
> +
> +def summarystat(listd):
> +    for i in range(len(listd)):
> +        avg=np.mean(listd[i])
> +        median=np.median(listd[i])
> +        stdev=np.std(listd[i])
> +        plt.hist(listd[i], normed=True, bins=20)
> +        plt.title('The "Normal" Distribution with Mean & St. Devs.')
> +        plt.xlabel("Variables"); plt.ylabel("Frequency")
> +        plt.grid(True)
> +        plt.rc('grid', linestyle="dashed", color='grey')
> +        plt.show()
> +        print(i,"Mean = {0}".format(avg),"Median = {0}".format(median),"stdev = {0}".format(stdev))
> +    return avg,median,stdev
>
> I think it would also be helpful to have the min/max values of each column
> so the user knows the range of each of their variables.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_EHWUSF_HS68-5F2018-5FProject-5F1_pull_32-23pullrequestreview-2D140867839&d=DwMCaQ&c=qgVugHHq3rzouXkEXdxBNQ&r=0cV_TVOWlRVep9Kq29fJkEZ32-TILX0rx9zYhsQYYog&m=3fZ8LvrpgOvOpQDSEildYnhz_VXywjVQjrXy_4fi618&s=DIDAvWMTwm_LjZQZbSauD3RY4iei5pzN-jmx5SCK7ME&e=>,
> or mute the thread
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_Ag-5FPINxqJ6nAtxW7l1TE13pjxZJoGI-2D1ks5uKhK7gaJpZM4VXYic&d=DwMCaQ&c=qgVugHHq3rzouXkEXdxBNQ&r=0cV_TVOWlRVep9Kq29fJkEZ32-TILX0rx9zYhsQYYog&m=3fZ8LvrpgOvOpQDSEildYnhz_VXywjVQjrXy_4fi618&s=4zOM-ZOqp6sbh7mbQCwnByJjNVuuglI2H7JzWzW6Z0Q&e=>
> .
>

nitieaj commented:
typpe is refering to the categorical variable in the dataset "df"

On Mon, Jul 23, 2018 at 3:43 PM, haleyhowe <notifications@github.com> wrote:

> *@haleyhowe* commented on this pull request.
> ------------------------------
>
> In hs628_tools/Preprocess_module.py
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_EHWUSF_HS68-5F2018-5FProject-5F1_pull_32-23discussion-5Fr204575884&d=DwMCaQ&c=qgVugHHq3rzouXkEXdxBNQ&r=0cV_TVOWlRVep9Kq29fJkEZ32-TILX0rx9zYhsQYYog&m=kFDhEnkQq9uD6iSOJnO_09OjeTKjkoF8YpROMXAT_rY&s=7OPPUdXFU-qDlZIu-DihW4jHVHF_A3zRfG5iyn_LkG0&e=>
> :
>
> > +
> +
> +# In[239]:
> +
> +
> +df1["HNR_status"] =list(zip(df1.HNR, df1.status))
> +df1["HNR_status"].head(10)
> +
> +
> +# Build a tuple of the column and class label
> +
> +# In[240]:
> +
> +
> +
> +def coltuple(df,col,typpe):
>
> What is the parameter "typpe" referring to ?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_EHWUSF_HS68-5F2018-5FProject-5F1_pull_32-23pullrequestreview-2D139680900&d=DwMCaQ&c=qgVugHHq3rzouXkEXdxBNQ&r=0cV_TVOWlRVep9Kq29fJkEZ32-TILX0rx9zYhsQYYog&m=kFDhEnkQq9uD6iSOJnO_09OjeTKjkoF8YpROMXAT_rY&s=mEYFzs0EqxCMp76KMYMRF1v9zBgwKNOYk81pD5OO57Q&e=>,
> or mute the thread
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_Ag-5FPIEt5UExkL9QVD9Lb9nVp3P-5FX-5FEmJks5uJlGMgaJpZM4VXYic&d=DwMCaQ&c=qgVugHHq3rzouXkEXdxBNQ&r=0cV_TVOWlRVep9Kq29fJkEZ32-TILX0rx9zYhsQYYog&m=kFDhEnkQq9uD6iSOJnO_09OjeTKjkoF8YpROMXAT_rY&s=tlRzT9ldBxpvRCzNohoEMKlu4pepTfbV67AKbyrJAC0&e=>
> .
>

nitieaj committed:
Additional function descriptions and labels
 Changes to be committed:
 new file:   Preprocess1_module.py

nitieaj committed:
- Updates to the documentation of the methods

  Ongoing build

hhan14 commented on hs628_tools/Preprocess1_module.py:
```diff
@@ -9,8 +9,33 @@
+        """
+               #Returns the three input parameters
+               col_cat=(input("what is col_cat name ?"))
+               col=float(input("what is the col name ?"))
```
Hi Nitie, I just tried running your work for review on my PyCharm and got the error message as below:
File "<input>", line 35
    col_cat=(input("what is col_cat name ?"))
    ^
IndentationError: unexpected indent

Just wanted to suggest you revisit your work to confirm debugging

hhan14 replied:
![indentationerror](https://user-images.githubusercontent.com/35585357/43371845-c3a34b64-93d4-11e8-8057-d6d5d44f4ba1.png)
Hope this attached hleps!

RoxanneXin commented on hs628_tools/Preprocess1_module.py:
```diff
@@ -9,8 +9,33 @@
+    df1, col, col_cat = getInputs()
+    listgrp=coltuple(df1,col,col_cat)# helper function
+    colgroupbyclass=group_by(listgrp,1) # helper function
+    print(summarystat(colgroupbyclass))
```
Hi Nitie~ I think most main functions are usually placed at the end in module. But it will be fine if you keep calling the main at the end.

nitieaj committed:
- documentation of working methods and functions

- Working to wrap stand-alone functions in appropriate class that can be used in the general class module.

nitieaj committed:
- methods wrapped in a class

- Updated functions in the viz class

EHWUSF commented:
Having 4 different versions of the file still being present in this PR makes it a nuisance to even figure out which file really represents the current state of the PR, and of course it could not be merged like this.  So please remove the superfluous files.  Also, by this time the work-in-progress label should be removed (or at the very least some explanation given about why it is remaining).

The 4 different file versions also means that comments here don't necessarily get marked as outdated even if you've already changed the code they referred to, and the git history for the final file will only be a subset of the actual history.  It could be possible to clean that up by basically recreating the sequence of development, applying all the stepwise changes to a single file and making corresponding commits as you go (but given that we are tight on time now I won't recommend that here),  

It looks to me like Preprocess3_module.py is the name of the file used in the most recent commit, so that's the one I'll review.

EHWUSF commented on hs628_tools/Preprocess3_module.py:
```diff
@@ -0,0 +1,63 @@
+
+class vizsumstat:
+
+        def __init__ (self,csvPath,colname,col_catname):
```
Although I had said I would not be specifically noting all PEP 8 issues, some stylistic conventions are important enough that I will note them, like here-- there should not be any spaces after the function name and before the parenthesis.  There should also be a space after each of the commas here.  And there really should be a docstring explaining the significance of the nonstandard parameters (and I rather suspect they could be named more clearly also).

nitieaj replied:
Corrections added

EHWUSF commented on hs628_tools/Preprocess3_module.py:
```diff
@@ -0,0 +1,63 @@
+
+        def __init__ (self,csvPath,colname,col_catname):
+
+                self.df = pd.read_csv(csvPath)
```
As I noted in class, it's best to have your code simply require that the user pass the data as an object rather than read it in yourself.  There are a variety of reasons that their data might not be in a local file.  Also this project was supposed to avoid relying on pandas.

EHWUSF commented on hs628_tools/Preprocess3_module.py:
```diff
@@ -0,0 +1,63 @@
+                self.colgroupbyclass=self.group_by(self.tuple)
+
+                
+        def group_by(self,tpls,idx=1,merge=True):
```
The function needs a docstring explaining its purpose and parameters, and the code itself needs comments.

nitieaj replied:
docstring added

EHWUSF commented on hs628_tools/Preprocess3_module.py:
```diff
@@ -0,0 +1,63 @@
+                        d.update({k:v})
+                return d
+
+        def coltuple(self,df,col,typpe):
```
This function needs a doscstring explaining its purpose.  It also looks like the loops are currently unnecessary, that the entire body of the function could be replaced with a single line.

nitieaj replied:
docstring added

EHWUSF commented on hs628_tools/Preprocess3_module.py:
```diff
@@ -0,0 +1,63 @@
+                                #print (newtuple)
+                        return newtuple
+                
+        def summarystat(self):
```
Needs a docstring.

nitieaj replied:
Docstrings included

EHWUSF commented on hs628_tools/Preprocess3_module.py:
```diff
@@ -0,0 +1,63 @@
+                self.col_catname = col_catname
+                #print(self.colname)
+                self.tuple = self.coltuple(self.df, self.colname, self.col_catname)
+                self.colgroupbyclass=self.group_by(self.tuple)
```
this attribute name should use underscores to improve readability, and the ```=``` operator here should have a space before and after.

EHWUSF commented on hs628_tools/Preprocess3_module.py:
```diff
@@ -0,0 +1,63 @@
+                        return newtuple
+                
+        def summarystat(self):
+                for i in range(len(self.colgroupbyclass)):
```
It looks to me like the sole use of ```i``` here is as an unmodified index into ```self.colgroupbyclass```, and so it would be cleaner to instead just loop over the groups directly:
```
    for group in self.colgroupbyclass:
        avg=np.mean(group)
```

It also seems to me that these plots would be much more helpful as side-by-side subplots instead of being sequentially displayed, but I am also strongly suspecting I'm not fully understanding the intention here, due to the dearth of description and comments, and that the plots currently seem somewhat mislabeled to me.

EHWUSF commented on hs628_tools/Preprocess3_module.py:
```diff
@@ -0,0 +1,63 @@
+                       median=np.median(  self.colgroupbyclass[i])
+                       stdev=np.std(  self.colgroupbyclass[i])
+                       plt.hist(  self.colgroupbyclass[i], normed=True, bins=20)
+                       plt.title('The "Normal" Distribution with Mean & St. Devs.')
```
Is this really the right title?  I don't think that these distributions are necessarily 'Normal', and although you print out the mean and standard deviation, outside of a Jupyter notebook that info will be displayed quite separately from the displayed plots.  The plot titles should probably also mention what the chosen columns were, and something that distinguishes the various plots displayed.

EHWUSF commented:
This PR description is a bit too terse (and should also include a link back to the Issue it was meant to address).  Please clarify the description.

EHWUSF commented on hs628_tools/Preprocess3_module.py:
```diff
@@ -0,0 +1,63 @@
+                d = dict()
+                for tpl in tpls:
+                        k = tpl[idx]
+                        v = d.get(k,tuple()) + (tpl[:idx]+tpl[idx+1:] if merge else (tpl[:idx]+tpl[idx+1:],))
```
Especially be sure to clearly explain what this line is doing.

EHWUSF commented on hs628_tools/Preprocess3_module.py:
```diff
@@ -0,0 +1,63 @@
+                       stdev=np.std(  self.colgroupbyclass[i])
+                       plt.hist(  self.colgroupbyclass[i], normed=True, bins=20)
+                       plt.title('The "Normal" Distribution with Mean & St. Devs.')
+                       plt.xlabel("Variables"); plt.ylabel("Frequency")
```
I am not making sense of "Variables" as the name for the x axis here.

Also, if I download a local copy of the UCI parkinson's database, saving it in the hs628_tools directory as "parkinsons_train_set.csv", and then adjust the filename below to just refer to the local copy rather than requiring your particular path structure, and then run this script specifying "HNR" and "status", then this histogram gets displayed with the first column having a frequency over 100, which suggests there is probably something wrong with your ```group_by``` method.

EHWUSF commented on hs628_tools/Preprocess3_module.py:
```diff
@@ -0,0 +1,63 @@
+                return avg,median,stdev
+
+def main():
+        col=(input("input col name"))
```
it's best to end your prompt with a colon and a space, as you do below.  Also, when you are just wanting plain string input as you are here, it is best to use ```raw_input``` rather than ```input```, or else explain in your prompt that the user will need to provide quotation marks around their input (or the program will crash unless they happen to give a name that matches one of your variables).

And the outer parentheses here are just clutter.

EHWUSF commented on hs628_tools/Preprocess3_module.py:
```diff
@@ -0,0 +1,63 @@
+def main():
+        col=(input("input col name"))
+        col_catname=(input("inputcategorical col name: "))
+        path  = "C:/Users/Nitie/Desktop/health_informatics/MSHI_CLASSES/summer_2018/650/project data/parkinsons_train_set.csv"
```
Avoid hardcoding a path that is virtually guaranteed to make the program crash if someone else tries to test it as-is-- it's best to take the filename from the command line, or make it a local file.

nitieaj replied:
I have updated the module with your comments.Thanks

nitieaj committed:
-Comment and corections implemented.
 -Some of the functions & plots would  be updated with dynamic X-label
  capabilities.
